shaders/palette.frag

```diff
@@ -1,17 +1,36 @@
 #version 330 core
 
 in vec2 tex_coord;
+in vec2 fbo_coord;
+in float blend_mode;
 layout (location = 0) out vec4 color;
-layout (std140) uniform props {
-    int mode;
+uniform sampler2D atlas;
+uniform sampler2D framebuffer;
+layout (std140) uniform remaps {
+    vec4 mappings[19 * 256];
 };
 
-uniform sampler2D image;
-
 void main() {
-    vec4 texel = texture(image, tex_coord);
-    if (texel.g == 0) {
-        discard;
+    // Get rid of the branching later! We can split this into two or three shaders pretty easily
+    // TODO: FIX LATER.
```

Comment:
Also sampling textures with implicit gradients in divergent control flow is UB. See https://themaister.net/blog/2019/09/12/the-weird-world-of-shader-divergence-and-lod/

You might get away with this in a fullscreen pass but generally it can result in really funky artifacts

Reply:
Yeppers. In any case this particular shader is going to go through a ton of new stuff while we try to get the original games crap sorted out. It will probably end up in a couple of different vertex shaders and a larger bunch of fragment stuff.

Reply:
Shaders make it pretty easy to test out stuff :)